### ISSUE TITLE
Calendar Provider

### ISSUE THREAD

**Issue by alensiljak** [orgzly-revived/orgzly-android-revived#39]
This is a long-term dream of having the task dates displayed in a Calendar.
Orgzly should register itself as a calendar provider, allowing regular calendar apps to display the timed events visually on a time calendar.

The first calendar to offer would be Agenda. Agenda is a pre-defined query, whose results could be offered through the Agenda "calendar". Agenda is limited to the upcoming 2 weeks and that might be enough for now, to see the most-immediate upcoming tasks.

Other calendars can be worked-out later. Generally, any saved query could be offered as a calendar, too.

Not all tasks have dates set so this needs to be fleshed-out, if required at all. 

Any calendar app on the device can choose to display this calendar along with others, "real" calendars, or just in a week, day, or month view.

This is the ultimate goal and overshadows #38, which is just a step in that direction. A lot of work overlaps, anyway, so #38 might also be possible as a side-effect, if this is implemented.

Related: 
- https://github.com/orgzly/orgzly-android/issues/944
- #38 
- https://github.com/orgzly/orgzly-android/issues/57

**Comment by Ypot** [comment]
Orgzly has an agenda that I think is not limited to 2 weeks.

Try it using this search:
(b.notebook) o.e o.s ad.35

That would be the agenda for 35 days


For me out would be too a very needed feature.

I suppose the calendar should be generated automatically, maybe like the widget.
And probably orgzly should create an ID for each entry, if it doesn't exist.

**Comment by alensiljak** [comment]
I did not mean for Orgzly to implement the calendar widget but just to offer saved searches through a Calendar Provider. The display, reminders, and all the other calendar features have been implemented many times over by numerous calendar apps, which I'm sure we all already have.
By implementing the Calendar Provider, any calendar would be able to display it or act upon it, show widgets or anything else the calendars do.
There's too many good calendar apps out there. No need to waste effort reinventing the wheel.

**Comment by alensiljak** [comment]
Right. I mentioned ICSx5 a couple of times in https://github.com/orgzly/orgzly-android/issues/57. I've been using it for years for DAV sync in general (contacts and calendars).

https://github.com/larkery/orgcal is also worth looking into. It might be simpler. 

But both projects act as providers, so the code is "out there".

**Comment by stfl** [comment]
+1

**Comment by ShuoShen** [comment]
@alensiljak Love it.  On top of that, it would also be very nice to have Calendar sync because likely we'll need to update time and events on both ends.

**Comment by Ypot** [comment]
Could this be of any use?
https://codeberg.org/felipeqq2/orgtoical

**Comment by amberin** [comment]
Closing as implemented in #834. Further improvements should be handled as new issues. Thanks again, @anaxonda! 🎉
